browser-test/src/admin_predicates.test.ts

```diff
@@ -125,5 +125,78 @@ describe('create and edit predicates', () => {
     expect(await page.innerText('#application-view')).toContain('Screen 2');
 
     await endSession(browser);
-  })
+  });
+
+  it('every right hand type evaluates correctly', async () => {
+    const { browser, page } = await startSession();
+
+    await loginAsAdmin(page);
+    const adminQuestions = new AdminQuestions(page);
+    const adminPrograms = new AdminPrograms(page);
+
+    // DATE, STRING, LONG, LIST_OF_STRINGS, LIST_OF_LONGS
+    await adminQuestions.addNameQuestion('single-string');
+    await adminQuestions.addTextQuestion('list of strings');
+    await adminQuestions.addNumberQuestion('single-long');
+    await adminQuestions.addDateQuestion('date');
+    await adminQuestions.addCheckboxQuestion('both sides are lists', ['dog', 'rabbit', 'cat']);
+    await adminQuestions.addTextQuestion('depends on previous');
+
+    const programName = 'test all predicate types';
+    await adminPrograms.addProgram(programName);
+    await adminPrograms.editProgramBlock(programName, 'string', ['single-string']);
+    await adminPrograms.addProgramBlock(programName, 'list of strings', ['list of strings']);
+    await adminPrograms.addProgramBlock(programName, 'long', ['single-long']);
+    await adminPrograms.addProgramBlock(programName, 'date', ['date']);
+    await adminPrograms.addProgramBlock(programName, 'two lists', ['both sides are lists']);
+    await adminPrograms.addProgramBlock(programName, 'last', ['depends on previous']);
+
+    // Simple string predicate
+    await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 2');
+    const adminPredicates = new AdminPredicates(page);
+    await adminPredicates.addPredicate('single-string', 'shown if', 'first name', 'is not equal to', 'hidden');
+
+    // Single string one of a list of strings
+    await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 3');
+    await adminPredicates.addPredicate('list of strings', 'shown if', 'text', 'is one of', 'blue, green');
+
+    // Simple long predicate
+    await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 4');
+    await adminPredicates.addPredicate('single-long', 'shown if', 'number', 'is equal to', '42');
+
+    // Date predicate
+    await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 5');
+    await adminPredicates.addPredicate('date', 'shown if', 'date', 'is earlier than', '2021-01-01');
+
+    // Lists of strings on both sides (multi-option question checkbox)
+    await adminPrograms.goToEditBlockPredicatePage(programName, 'Screen 6');
+    await adminPredicates.addPredicate('both sides are lists', 'shown if', 'selections', 'contains any of', 'dog,cat');
+
+    await adminPrograms.publishProgram(programName);
+
+    // Switch to applicant view - if they answer each question according to the predicate,

```

Comment:
praise: Great test - and clever solution here to try to compact the test as much as possible while still thoroughly testing behavior.

Reply:
Thanks! I tried :)